common/frame_session.go

```diff
@@ -355,7 +355,13 @@ func (fs *FrameSession) parseAndEmitWebVitalMetric(object string) error {
 	state := fs.vu.State()
 	tags := state.Tags.GetCurrentValues().Tags
 	if state.Options.SystemTags.Has(k6metrics.TagURL) {
-		tags = tags.With("url", wv.URL)
+		if name, ok := fs.page.urlGroupingName(fs.ctx, wv.URL); ok {
+			tags = tags.With("url", name)
+			tags = tags.With("name", name)
+		} else {
+			tags = tags.With("url", wv.URL)
+			tags = tags.With("name", wv.URL)
+		}
```

Comment:
Should we put this part into a function because we duplicate it thrice?

Reply:
Nice idea. Done in 70c9de8a640b77035f016a2f6e5a1f9b1aa2eacf

Reply:
👍